### ISSUE TITLE
Questions regarding 03 - Calling-an-API 

### ISSUE THREAD

**Issue by kurtzilla** [auth0-samples/auth0-react-samples#14]
I am running into some problems running the example and not sure if I am setting them up correctly. 

Firstly, I am not sure what to use for the AUTH0_AUDIENCE variable - I am using the value listed in my Api tab and am not sure if this is correct
![screen shot 2017-06-16 at 12 32 33 pm](https://user-images.githubusercontent.com/579932/27239930-293359d8-5290-11e7-893a-840423b17eb7.png)

However, I am able to login, view my profile and ping the public endpoint. 

It is when I try to call the private endpoint that I receive the message that I am 'unauthorized'

Could you provide any assistance on what I may be doing wrong?

Thank you!

**Comment by chenkie** [comment]
You've got the correct value in for your audience. Can you show me your `WebAuth` config? Also, can you show me the tokens you have in local storage?

**Comment by kurtzilla** [comment]
Where do I find the webAuth config? Is that somewhere within the Auth0 dashboard?

access_token: eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlJqUkNSRGd3TkRNMlJrSkVRVVJDTlRjeU1ETTBOalZEUVVGRFJEVXpNRVF3T1RrNE5qQXhNUSJ9.eyJpc3MiOiJodHRwczovL2t1cnR6LmF1dGgwLmNvbS8iLCJzdWIiOiJmYWNlYm9va3wxNTAwNjAwMjU2NjM3NDQ4IiwiYXVkIjpbImh0dHBzOi8va3VydHouYXV0aDAuY29tL2FwaS92Mi8iLCJodHRwczovL2t1cnR6LmF1dGgwLmNvbS91c2VyaW5mbyJdLCJhenAiOiJJRWlVdVM2Um40clFGUWd4NlVPdFJNa3FTTTRPTFN2SSIsImV4cCI6MTQ5NzY0NTU5MywiaWF0IjoxNDk3NjM4MzkzLCJzY29wZSI6Im9wZW5pZCBwcm9maWxlIn0.Ou2TVxoLH6Jwdi7-vdcoa4Wf1fQckUOz1XCQmKgq0haZLEHs5-qi9MEuu6h8qm624WAv9CgR9b61jmWfUC7Uv2LfXudgG9rQqN8JSzlQILEmfPHCCVdzZFkTTai-Uog_Si-sx4ZVfvvAJyfq5QTmcPinrpNBBSxxPWCfwZ8C7GWYJvxWYoImu9SX9o1mD5D-oe3BlihYLmJ4QyQcjnhH1C04fmmL2lUBKJUQvTaJ_YS5tvxfSL0lEvRwIq8H1XtKbYuBZ1u7PegzAF8uBpGT-KGBQEbi_PIM4gMdBVh2Ezl4w0MIpnOrEMwGw5PyTu2FN6PUxzwLOcKLNmDvIDYIhw

id_token: eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IlJqUkNSRGd3TkRNMlJrSkVRVVJDTlRjeU1ETTBOalZEUVVGRFJEVXpNRVF3T1RrNE5qQXhNUSJ9.eyJuYW1lIjoiUm9iIEt1cnR6IiwiZ2l2ZW5fbmFtZSI6IlJvYiIsImZhbWlseV9uYW1lIjoiS3VydHoiLCJuaWNrbmFtZSI6InJvYiIsInBpY3R1cmUiOiJodHRwczovL3Njb250ZW50Lnh4LmZiY2RuLm5ldC92L3QxLjAtMS9jOS4wLjUwLjUwL3A1MHg1MC8xMDM3NjMxMV83NjE1NTIyOTA1NDIyNTJfODk2MzgwOTA2NjM1NDkxNzc4NV9uLmpwZz9vaD1jYTRiMDI2MzM0MWQ1M2RlYThhYzc5MWRmMmEyMDUyOCZvZT01OUU4MkM0MyIsImdlbmRlciI6Im1hbGUiLCJsb2NhbGUiOiJlbi1VUyIsInVwZGF0ZWRfYXQiOiIyMDE3LTA2LTE2VDE4OjM5OjUyLjg5NVoiLCJpc3MiOiJodHRwczovL2t1cnR6LmF1dGgwLmNvbS8iLCJzdWIiOiJmYWNlYm9va3wxNTAwNjAwMjU2NjM3NDQ4IiwiYXVkIjoiSUVpVXVTNlJuNHJRRlFneDZVT3RSTWtxU000T0xTdkkiLCJleHAiOjE0OTc2NzQzOTMsImlhdCI6MTQ5NzYzODM5Mywibm9uY2UiOiJOZVFnYzRKb2hMeVlYZmJoVFJseXZvTjg2THBjVUlrciIsImF0X2hhc2giOiJLdnM4SEt0WFpUQkJGcFRkVms1Vm1BIn0.dxs0tqGJ8SSGejSkeIDKBvOvl00M4jMFQy39AfjbSNmSrKiSLzVRkMGKAbInKau1MeIE9lOgLLVoudeuAy6N34FINSa1unIbffXoRS9e5w23N1UvxLuFuhzrAUUwVvVa34PxwzO682hALKZbqPc4RNdq2D5duOoI4YpZBAtLiwiGLSAsBaucLMpT5gNKdZJLqmf1dYjKBvBsk48ummqUyjktljpgG5_R3cEITNR6qDB1wPs3G8PUhFD24la_oBRq__eMmvCyHnZTIZ5sA95HnflitmF5hb3E7AU_ESUZYtK2b0xg1YxGrL8FnrPoDoy0KQtQNqcleZdj5E7iwCF1oA

**Comment by thameera** [comment]
@kurtzilla Are you using the same audience in Lock and in your backend? (`.env` file)

I'd suggest that you create a new API for your backend in the Auth0 dashboard (in APIs section) and use the identifier you specify as the audience in both Lock and backend.

**Comment by kurtzilla** [comment]
I am new to all of this and might need a bit more hand-holding to understand everything. 

Could someone explain the purpose of "audience" for me? I think I am confused as to how all the pieces go together.

.env file
```
AUTH0_AUDIENCE=https://xxxxx.auth0.com/api/v2/
AUTH0_DOMAIN=xxxxx.auth0.com
```

auth0-variables.js
```
export const AUTH_CONFIG = {
  domain: 'xxxxx.auth0.com',
  clientId: 'IEiUuS6Rn4rQFQgx6UOtRMkqSM4OLSvI',
  callbackUrl: 'http://localhost:3000/callback',
  apiUrl: 'https://xxxxx.auth0.com/api/v2/'
}
```

**Comment by chenkie** [comment]
Hey @kurtzilla,

You'll need to creat a new API in the APIs section of the dashboard. Give it an identifier. It can be anything you like, but a good choice is the URL for your actual API. 

Use this identifier as your `AUTH0_AUDIENCE`. When you set up the configuration for `Auth0.WebAuth` in your app (assuming you are following the 03 - Calling an API sample directly), pass this audience as the `audience` key. 

This will get you a JWT access token when you authenticate which is what you want. 

Next you can use the same audience value to protect your API. The sample includes a sample Node.js API but you can follow the Backend/API quickstarts to see how to protect an API for the specific technology you are using.

**Comment by thameera** [comment]
Audience specifies which API your access token is intended to call.

In your case, the audience is your backend API. But since Auth0 does not know about it specifically, you need to create a new 'API' by going to Auth0 dashboard > APIs > Create API. You can give your website's URL, etc as the identifier.

In this specific example, you will need to add the `read:messages` scope to the API as it's checked in `server.js`. We'll update the README to cover all these steps, since this seems to be a bit confusing to new users.

Here's a ~5 minute screencast that goes through all these steps: https://www.youtube.com/watch?v=ti2zMJm34Cw

**Comment by kurtzilla** [comment]
Thank you both for your assistance.
I was able to get this working locally (api at http://localhost:3001) which was what I was hoping to do originally. I figured I could work on configuration of a publicly deployed api once I had established a solution on my dev machine. I think I am getting a better grasp of the layout and the screencast was helpful.

Also, I had been using a default API (?) created at Auth0 and that seemed to be causing some issues as well - ie: I was unable to add in the `read:messages` permission. I created a brand new API and set its identifier to `http://localhost:3001`, added a new scope with the name `read:messages`, and updated the .env and auth0-variables.js files in the provided solution to use the http://localhost:3000 url. (**noting that constants.js did not need any changes in this particular case).

@chenkie I was still confused as to the reference to auth0.WebAuth, but now I see you are referring to the call within the Auth/Auth.js file that sets up the auth0 component:
```
export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience: AUTH_CONFIG.apiUrl,
    responseType: 'token id_token',
    scope: 'openid profile read:messages'
  });
...
```

Thanks again
